ignite/engine/__init__.py

```diff
@@ -83,16 +85,19 @@ def supervised_training_step(
         trainer = Engine(update_fn)
 
     .. versionadded:: 0.4.5
+    .. versionadded:: 0.5.0
+        Added Gradient Accumulation.
     """
 
     def update(engine: Engine, batch: Sequence[torch.Tensor]) -> Union[Any, Tuple[torch.Tensor]]:
         model.train()
-        optimizer.zero_grad()
         x, y = prepare_batch(batch, device=device, non_blocking=non_blocking)
         y_pred = model(x)
-        loss = loss_fn(y_pred, y)
+        loss = loss_fn(y_pred, y) / gradient_accumulation_steps
```

Comment:
```suggestion
        loss = loss_fn(y_pred, y) / gradient_accumulation_steps if gradient_accumulation_steps > 1 else loss_fn(y_pred, y)
```
to avoid additional useless kernel execution

Reply:
Okay I have added the check as a seperate line, flake8 was making it a little less readable if we kept it like this in the same line.

Reply:
Also I forced the commit by mistake which caused your commit to be overwritten, I have merged the master again sorry.

Reply:
Not sure that it adds overhead. The AD should detect a division by a scalar so no impact on the derivatives and only one extra division on the value.

Reply:
It's about performances (not AD) and skip +1 CUDA kernel execution